app/controllers/admin/gdpr_delete_requests_controller.rb

```diff
@@ -3,7 +3,10 @@ class GDPRDeleteRequestsController < Admin::ApplicationController
     layout "admin"
 
     def index
-      @gdpr_delete_requests = ::GDPRDeleteRequest.order(created_at: :desc).page(params[:page]).per(50)
+      @gdpr_delete_requests = Admin::GDPRDeleteRequestsQuery.call(relation: ::GDPRDeleteRequest.all,
+                                                                  options: params.permit(
+                                                                    :search,
```

Comment:
We're not passing the `:role` to the query class.  My assumption is that you are writing future functionality but we don't have the UI component for this.